components/PostImage/PostImage.js

```diff
@@ -1,24 +1,55 @@
 import styles from './PostImage.module.css';
 import cx from 'classnames';
+import React from 'react';
 
 export function PostImage({
   imageURL,
   fraction,
   aspectRatio,
   isLoading,
   setIsLoading,
+  onDoubleClick,
 }) {
+  const [isImgDoubleClicked, setIsImgDoubleCLicked] = React.useState(false);
+  console.log({ isImgDoubleClicked });
   return (
-    <img
-      className={cx(styles.postImg, {
+    <div
+      onDoubleClick={() => {
+        setIsImgDoubleCLicked(true);
+        setTimeout(setIsImgDoubleCLicked, 1500, false);
+      }}
+      width={`${fraction * 100}%`}
+      className={cx(styles.imgContainer, {
         [styles.isLoading]: isLoading,
       })}
-      src={imageURL}
-      alt="post"
-      loading="lazy"
-      width={`${fraction * 100}%`}
-      style={{ aspectRatio }}
-      onLoad={() => setIsLoading?.(false)}
-    />
+      style={{ aspectRatio, width: `${fraction * 100}%` }}
+    >
+      <img
+        className={cx(styles.postImg, {
+          [styles.isLoading]: isLoading,
+        })}
+        src={imageURL}
+        alt="post"
+        loading="lazy"
+        style={{ aspectRatio }}
+        onLoad={() => setIsLoading?.(false)}
+        onDoubleClick={onDoubleClick}
+      />
+      {isImgDoubleClicked && (
+        <div className={styles.likeIconContainer}>
+          <svg
+            className={styles.likeIcon}
+            aria-label="Unlike"
+            fill="white"
+            height="24"
+            role="img"
+            viewBox="0 0 48 48"
+            width="24"
```

Comment:
Instead of scaling, you can start with the right dimensions
```suggestion
            height="90"
            role="img"
            viewBox="0 0 48 48"
            width="90"
```